### ISSUE TITLE
sort errors out with SEGV

### ISSUE THREAD

**Issue by briantigerchow** [samtools/samtools#611]
Recently, after hours of running, five separate runs of samtools 1.3.1 errored out with SEGV. I invoked the following command:

```
samtools sort -@ 12 -m 3G -o sorted_bam_path bam_path
```

**Comment by tseemann** [comment]
@briantigerchow 
how much RAM does your compute node have?
did you compile from source?

**Comment by briantigerchow** [comment]
Yes. Compiled from source. Here is the command I used to install:

``` make
env/samtools/samtools:
        git clone --branch 1.3.1 https://github.com/samtools/samtools.git env/samtools || true
        rm -rf env/samtools/.git
        cd env/samtools && make
```

As you can see, I cannot provide the git commit hash of the installation.

RAM varies. Three of the five jobs were on 64 GB nodes. Two were on ~378 GB nodes. 

12 cores and 36 GB allocated to each of the jobs with 4.0-4.5 GB in use at time of crash.
